File: src/main/java/com/zndevs/toolset/commands/ToolSetMainCommand.java
package com.zndevs.toolset.commands;

import com.zndevs.toolset.ToolSetCommand;
import com.zndevs.toolset.ToolSetConfiguration;
import com.zndevs.toolset.tools.CommandResponse;
import com.zndevs.toolset.tools.ToolSetOptions;
import org.bukkit.Bukkit;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * Base ToolSet Command
 */

public class ToolSetMainCommand implements ToolSetCommand {
    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args) {
        CommandResponse response = new CommandResponse();

        if (args.length > 0) {
            List<String> argsList = new ArrayList<>(Arrays.asList(args));
            argsList.subList(0, 1).clear();
            String[] argsCut = argsList.toArray(new String[argsList.size()]);

            switch (args[0]) {
                case "update":
                case "u":
                    new ToolSetUpdateCommand().onCommand(sender, command, label, argsCut);
                    break;
                case "reload":
                case "r":
                    new ToolSetReloadCommand().onCommand(sender, command, label, argsCut);
                    break;
                default:
                    response.setPlayerMessage(ToolSetOptions.getHelpFormat("ToolSet", new String[]{
                            "update, u",
                            "reload, r"
                    }));
                    break;
            }
        } else {
            response.setPlayerMessage(ToolSetOptions.getHelpFormat("ToolSet", new String[]{
                    "update, u",
                    "reload, r"
            }));
        }

        response.send(sender);

        return true;
    }

    @Override
    public ToolSetConfiguration setConfigurationDefaults(ToolSetConfiguration defaults) {
        // do nothing
        return defaults;
    }
}

File: src/main/kotlin/com/zndevs/toolset/commands/ToolSetMainCommand.kt
package com.zndevs.toolset.commands

import com.zndevs.toolset.ToolSetCommand
import com.zndevs.toolset.ToolSetConfiguration
import com.zndevs.toolset.commands.tools.SimpleCommand
import com.zndevs.toolset.commands.tools.SubCommand
import com.zndevs.toolset.tools.ToolSetCommandTools
import org.bukkit.permissions.Permission
import org.bukkit.plugin.java.JavaPlugin

class ToolSetMainCommand : ToolSetCommand() {
    private var plugin: JavaPlugin? = null

    private val subCommands: Array<SubCommand> = arrayOf(
            SubCommand("command", arrayOf("c"), "Update or check the status of commands", ToolSetCommandCommand()),
            SubCommand("update", arrayOf("u", "upd"), "Update ToolSet modules or check if updates are required", ToolSetUpdateCommand()),
            SubCommand("reload", "Reload ToolSet and its modules", SimpleCommand("ToolSet Reloader", fun(_, _, _, _, _, r): Boolean {
                r.playerMessage = ToolSetCommandTools.reloadCommands(null)
                return true
            }, Permission("toolset.core.admin.reload")))
    )

    override fun getSubCommands(): Array<SubCommand> = subCommands
    override fun getCommandName(): String = "ToolSet"
    override fun setConfigurationDefaults(defaults: ToolSetConfiguration): ToolSetConfiguration = defaults
    override fun setPlugin(plugin: JavaPlugin) {
        this.plugin = plugin
    }
}